### PR TITLE
Upgrade Scala dependencies for 2.13 compatibility

### PR DESCRIPTION
**Pull request by cocreature**
This upgrades a bunch of Scala libraries to versions that have 2.13
support. There are two libraries that are still missing:

- diffson, this has a new version but with significant breaking
  changes and it is only used in Naigator console which I hope to kill
  before I have to worry about this.

- ai.x:diff, this is used in the ledger API test tool. The library is
  abondened but there are a few alternatives.

changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by SamirTalwar-DA** [comment]
I would be happy to kill the dependency on `ai.x:diff` in favor of Scalatest assertions (which you can now use without using the test runner).

**Comment by cocreature** [comment]
> I would be happy to kill the dependency on `ai.x:diff` in favor of Scalatest assertions (which you can now use without using the test runner).

Your choice but either way, that seems like something for a separate PR.

**Comment by SamirTalwar-DA** [comment]
Agreed, just thinking it'd help this one over the line.

**Comment by cocreature** [comment]
> Agreed, just thinking it'd help this one over the line.

Doesn’t matter much. If you look at the companion PR #8271, I’m starting out with just one target and we can gradually expand that adding the ledger API test tool later.

File: language-support/scala/bindings-akka/src/main/scala/com/digitalasset/ledger/client/binding/LedgerClientBinding.scala [hunk 1]
```diff
@@ -56,7 +56,7 @@ class LedgerClientBinding(
 
     logger.debug(
       "[tx {}] subscription start with offset template selector {}, start {}, end {}",
-      party,
+      Party.unwrap(party),
```
Comment (cocreature) [inline language-support/scala/bindings-akka/src/main/scala/com/digitalasset/ledger/client/binding/LedgerClientBinding.scala:59]: afaict, this is required as a side effect of https://github.com/scalaz/scalaz/pull/2076/files. @S11001001 Is there a better way of solving this?

**Comment by stefanobaghino-da** [comment]
> I would be happy to kill the dependency on `ai.x:diff` in favor of Scalatest assertions (which you can now use without using the test runner).

Scalatest assertions don't produce diffs and are very difficult to read.

It would make the life more difficult for users to interpret a test result.

I think it would be a good idea to preserve the diff feature.

Two possible alternatives to evaluate:
* [`softwaremill/diffx`](https://github.com/softwaremill/diffx) is a more recent alternative to `adotai/diff`
* MUnit assertions

I think the latter could be particularly interesting.

The former is probably a safer choice as it has more or less the same scope as `adotai/diff`.

**Review by aherrmann-da — APPROVED** [review]
LGTM

**Pull request merged**